internal/auth/credentials.go

```diff
@@ -0,0 +1,629 @@
+	methods := []string{deviceCodeTokenKey, authCodeTokenKey, clientCredentialsTokenKey}
+
+	for _, method := range methods {
+		storage := getTokenStorage(method)
+		if err := storage.ClearToken(); err != nil {
+			errs = append(errs, err)
+		}
+		// Also clear from file storage
+		if err := clearTokenFromFile(method); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// Also clear the cached PingOne API client to force re-initialization
+	ClearPingOneClientCache()
+
+	return errors.Join(errs...)
+}
+
+// ClearTokenForMethod removes the cached token for a specific authentication method
+// Clears from both keychain and file storage
+func ClearTokenForMethod(authMethod string) error {
+	var errs []error
+
+	// Clear from keychain
+	storage := getTokenStorage(authMethod)
+	if err := storage.ClearToken(); err != nil {
+		errs = append(errs, fmt.Errorf("keychain clear failed: %w", err))
+	}
+
+	// Also clear from file storage
+	if err := clearTokenFromFile(authMethod); err != nil {
+		errs = append(errs, fmt.Errorf("file clear failed: %w", err))
+	}
+
+	// Also clear the cached PingOne API client to force re-initialization
+	ClearPingOneClientCache()
+
+	return errors.Join(errs...)
+}
+
+// PerformDeviceCodeLogin performs device code authentication, returning the token, whether it's new authentication,
+// and any error encountered during the process
+func PerformDeviceCodeLogin(ctx context.Context) (*oauth2.Token, bool, error) {
+	cfg, err := GetDeviceCodeConfiguration()
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to get device code configuration: %w", err)
+	}
+
+	// Set grant type to device code
+	cfg = cfg.WithGrantType(svcOAuth2.GrantTypeDeviceCode)
+
+	// Load any existing cached token before calling SDK (checks both keychain and file storage)
+	tokenKey, err := GetAuthMethodKeyFromConfig(cfg)
+	var existingToken *oauth2.Token
+	if err == nil {
+		// Use LoadTokenForMethod which handles both keychain and file storage
+		existingToken, _ = LoadTokenForMethod(tokenKey)
+
+		// If we have a valid token, return it without performing new authentication
+		if existingToken != nil && existingToken.Valid() {
+			return existingToken, false, nil
+		}
+	}
+
+	// Get token source to perform authentication or use cached token
+	tokenSource, err := cfg.TokenSource(ctx)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to get token source: %w", err)
+	}
+
+	// Get token (SDK will return cached token if valid, or perform new authentication)
+	token, err := tokenSource.Token()
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to get token: %w", err)
+	}
+
+	// Save token using our storage method (respects --use-keychain flag)
+	if tokenKey != "" {
+		_ = SaveTokenForMethod(token, tokenKey)
+	}
+
+	// Determine if this was new authentication by comparing with what we loaded
+	newAuth := existingToken == nil || existingToken.AccessToken != token.AccessToken
```

Comment:
better to check existingToken.IsValid(), or set a boolean for this above where you get the existing token.

Same for the other methods